deps/src/main/java/com/microsoft/azure/sdk/iot/deps/auth/IotHubSSLContext.java

```diff
@@ -202,7 +188,7 @@ private void generateSSLContextWithKeys(String publicKeyCertificateString, Strin
         char[] temporaryPassword = generateTemporaryPassword();
 
         //Codes_SRS_IOTHUBSSLCONTEXT_34_020: [The constructor shall create a keystore containing the public key certificate and the private key.]
-        KeyStore keystore = KeyStore.getInstance(KeyStore.getDefaultType());
+        KeyStore keystore = KeyStore.getInstance("jks");
```

Comment:
We shouldn't hard code anyting here but instead rely on the `keystore.type` property. See the documentation for an explanation:

>public static final String getDefaultType​()
Returns the default keystore type as specified by the keystore.type security property, or the string "jks" (acronym for "Java keystore") if no such property exists.
>
>The default keystore type can be used by applications that do not want to use a hard-coded keystore type when calling one of the getInstance methods, and want to provide a default keystore type in case a user does not specify its own.
>
>The default keystore type can be changed by setting the value of the keystore.type security property to the desired keystore type.
>
>Returns:
>the default keystore type as specified by the keystore.type security property, or the string "jks" if no such property exists.
>See Also:
>security properties


https://docs.oracle.com/javase/9/docs/api/index.html?java/security/KeyStore.html

Reply:
Ok. Will change our properties. Thanx.